benchmarks/vector_db_compare/pgvector_bench.py

```
@@ -0,0 +1,417 @@
+        conn.execute(
+            f"create table {qualified_table} ("
+            f"doc_id integer primary key, "
+            f"id text not null, "
+            f"grp integer not null, "
+            f"embedding vector({dimensions}) not null)"
+        )
+        insert_start = time.perf_counter()
+        prepare_seconds = 0.0
+        copy_write_seconds = 0.0
+        with conn.transaction():
+            with conn.cursor().copy(f"copy {qualified_table} (doc_id, id, grp, embedding) from stdin") as copy:
```

Ruff (0.15.13)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)